### ISSUE TITLE
Deleting account results in internal server error

### ISSUE THREAD

**Issue by strk** [go-gitea/gitea#1145]
- Gitea version (or commit ref): 7ed00b6e
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes

## Description

1. Sign-in via (unknown) GitHub - delete link to existing GitHub if any
2. Register as new user
3. Sign-in again if needed (due to  #1144)
4. To to user settings
5. Delete account

Expected behavior: account is deleted, you're redirected to home.
Obtained behavior: internal server error, 500 page shown

\cc @willemvd

**Comment by lunny** [comment]
Could you take me more detail? What's step 2? What does `Register as new user` mean?

**Comment by lcges** [comment]
Really? It must be in full 1.1.0 version?
Put it in a 1.1.1 or 1.2.0 version

**Comment by lunny** [comment]
I cannot reproduce it. I maybe put it to v1.2 but in fact be back ported in v1.1.1

**Comment by strk** [comment]
"Register as new user" means that being the *first_time* you login
via GitHub (make sure it is) you are prompted to either *link* your
GitHub credential to an existing account *or* register a new account.
That's where you would pick "register new account" form.

**Comment by philfry** [comment]
I think that's what @strk meant (tested on try.gitea.io):

![screenshot-try gitea io-2017-03-08-10-20-44](https://cloud.githubusercontent.com/assets/1649785/23697744/413a4b36-03e9-11e7-8e26-d2ee1eae80eb.png)
![screenshot-try gitea io-2017-03-08-10-21-28](https://cloud.githubusercontent.com/assets/1649785/23697743/413a37ae-03e9-11e7-8cf7-fc7964ce6702.png)
![screenshot-try gitea io-2017-03-08-10-21-46](https://cloud.githubusercontent.com/assets/1649785/23697745/413ace3a-03e9-11e7-9254-20ade36c9464.png)
![screenshot-try gitea io-2017-03-08-10-22-01](https://cloud.githubusercontent.com/assets/1649785/23697747/413b6264-03e9-11e7-99d0-76828910ff59.png)
![screenshot-try gitea io-2017-03-08-10-22-13](https://cloud.githubusercontent.com/assets/1649785/23697746/413add30-03e9-11e7-9ef1-98f05c892a40.png)
![screenshot-try gitea io-2017-03-08-10-22-23](https://cloud.githubusercontent.com/assets/1649785/23697748/413e4236-03e9-11e7-9d07-35b859719b41.png)

**Comment by strk** [comment]
Exactly! Thanks Philippe :)

**Comment by lunny** [comment]
Thanks! I maybe follow this steps to reproduce it.

**Comment by lunny** [comment]
Thanks! I will follow this steps to reproduce it.

**Comment by lunny** [comment]
:( could you reproduce this on http://git.lunny.info ? still cannot reproduce it.

**Comment by pgaskin** [comment]
I have this issue too.

**Comment by lunny** [comment]
@geek1011 could you find the issue and send a PR to fix that?

**Comment by pgaskin** [comment]
@lunny I do not have much time, I'll see what I can do.

**Comment by lunny** [comment]
Thanks in advance

**Comment by pgaskin** [comment]
@lunny I think I have a few ideas about what might be happening. I just need to test them, which takes quite a bit of time due to the time it takes to set up this situation. I will not be able to look into this very much for the next few days due to a lot of other work to do. Also thanks to this project! I use daily for many different tasks.

**Comment by philfry** [comment]
re https://github.com/go-gitea/gitea/issues/1145#issuecomment-285241309
@lunny 
fwiw - I cannot reproduce the issue on your server.

**Comment by lunny** [comment]
Let's continue this on v1.2

**Comment by strk** [comment]
If it's a bug introduced by OAuth2 support it should be fixed in the
1.1 branch (the version that introduced OAuth2)

**Comment by lunny** [comment]
@strk I think so. But in fact 1.1 still have many bugs moved to 1.2 except this one. But we can't wait and wait. We can fix that if some one send a PR and publish 1.1.1 ASAP.

**Comment by ptman** [comment]
if we don't want bugs to delay 1.1.0, they should be moved to 1.1.x, not 1.2.0

features should be moved to 1.2.0

**Comment by lunny** [comment]
This bug will be fixed in 1.2.0 and then back ported 1.1.x. So we tag it 1.2.0

**Comment by strk** [comment]
I agree that Milestone would be more useful to be 1.1.x.
Or, every backport branch should be somehow encoded
(using labels, maybe ?)

**Comment by pgaskin** [comment]
I am kind of stuck on debugging this issue. Sometimes it happens, but other times it does not. 

If @strk has any ideas on the location, or if he thinks there is a more specific situation where this happens, it would be quite useful.

**Comment by strk** [comment]
@geek1011 set your logs level to Trace and check the logs, might contain some hints.
Being unable to reproduce is concerning, are you running the steps in a fresh environment always ?

**Comment by strk** [comment]
I could just reproduce on try.gitea.io running version 1.1.0+57-g9182a35f

To reproduce: 
  - Hit the "forgot password"
  - Open the link received by mail making sure to send no cookies (anonymous window)
  - Set the new password
  - Login with new credentials
  - Go to settings/delete_account
  - Enter the password

What happens: you get a 500 from the user profile page
What should happen: you are redirected to the homepage

**Comment by strk** [comment]
\cc @tboerger (if you have access to logs)

**Comment by strk** [comment]
Oh, and the user was NOT deleted (I see it still: https://try.gitea.io/strkhub)

**Comment by strk** [comment]
As I tried to reproduce locally with no luck, I'm thinking this is really due to the account being crated with OAuth2 LoginSource. That is the kind of bug that would be fixed by #1143

So renewed way to reproduce:
 1. Login via GitHub, creating a new account with it (github must not have been used before by any other user)
 2. Try to delete the account

Note that another side-effect of the LoginSource=OAuth2 is that you can NOT remove your GitHub association, so #1143 starts becoming urgent to merge !!!

**Comment by strk** [comment]
\cc @willemvd

**Comment by strk** [comment]
:( cannot reproduce like that either - so it takes looking at try.gitea.io logs to figure this out @tboerger

**Comment by strk** [comment]
So from the try.gitea.io logs:
```
2017/03/20 11:01:11 [...ters/user/setting.go:710 SettingsDelete()] [E] DeleteUser: ExternalLoginUser: database table is locked
```

The message comes from sqlite, used as backend on try.gitea.io.
@lunny mentioned he filed a PR that should fix this (which PR @lunny ?)

**Comment by strk** [comment]
I guess it is #1321

**Comment by lunny** [comment]
resolved by #1321

**Comment by strk** [comment]
Confirmed fixed now that v1.1.0-61-g888dee3 was deployed
